apps/modernization-ui/src/design-system/entry/multi-value/MultiValueEntry.tsx

```diff
@@ -0,0 +1,151 @@
+import { Icon } from '@trussworks/react-uswds';
+import { AlertBanner } from 'alert';
+import classNames from 'classnames';
+import { Button } from 'components/button';
+import { Heading } from 'components/heading';
+import { Column, DataTable } from 'design-system/table';
+import { ReactNode, useEffect, useState } from 'react';
+import { Control, DefaultValues, FieldValues, FormProvider, useForm } from 'react-hook-form';
+import styles from './MultiValueEntry.module.scss';
+
+type Props<V extends FieldValues> = {
+    title: string;
+    columns: Column<V>[];
+    defaultValues?: DefaultValues<V>; // Provide all default values to allow `isDirty` to function
+    errors?: string[];
+    onChange: (data: V[]) => void;
+    isDirty: (isDirty: boolean) => void;
+    formRenderer: (control: Control<V>) => ReactNode;
+    viewRenderer: (entry: V) => ReactNode;
+};
+export const MultiValueEntry = <V extends FieldValues>({
+    title,
+    defaultValues,
+    columns,
+    errors,
+    onChange,
+    isDirty,
+    formRenderer,
+    viewRenderer
+}: Props<V>) => {
+    const form = useForm<V>({ mode: 'onBlur', defaultValues });
+    const [data, setData] = useState<V[]>([]);
+    const [state, setState] = useState<'view' | 'edit' | 'new'>('new');
+    const [activeIndex, setActiveIndex] = useState<number | undefined>();
+
+    useEffect(() => {
+        onChange(data);
+    }, [data]);
+
+    useEffect(() => {
+        isDirty(form.formState.isDirty);
+    }, [form.formState.isDirty]);
+
+    const handleSubmit = () => {
```

Comment:
suggestion:  The call to form.values can be replaced by using the form's `handleSubmit` function.  It should accept a function that accepts `V`.